lib/token-store/index.js

```diff
@@ -236,10 +236,21 @@ module.exports = class TokenStore {
     /**
      * Gets the token starting at the specified index.
      * @param {number} offset - Index of the start of the token's range.
+     * @param {Object} [options=0] - The option object.
+     * @param {boolean} [options.includeComments=false] - The flag to iterate comments as well.
      * @returns {Token|null} The token starting at index, or null if no such token.
      */
-    getTokenByRangeStart(offset) {
-        const token = this.tokens[this.indexMap[offset]] || null;
+    getTokenByRangeStart(offset, options) {
+        const includeComments = options && options.includeComments;
+        const token = createCursorWithSkip(
+            cursors.forward,
+            this.tokens,
+            this.comments,
+            this.indexMap,
+            offset,
+            -1,
+            { includeComments }
+        ).getOneToken();
```

Comment:
`createCursorWithSkip` is the function to normalize options for overload of `number` (skip), `function` (filter), or `object`. I think we can use `cursors.forward.createCursor` directly since this does not use `skip`.

```js
const token = cursors.forward.createCursor(
    this.tokens,
    this.comments,
    this.indexMap,
    offset, // startLoc
    -1,     // endLoc
    includeComments,
    null,   // filter
    0,      // skip
    -1      // count
).getOneToken()
```

Or, it's nice if we define `cursors.forward.createBaseCursor` function with [this part](https://github.com/eslint/eslint/blob/591b74afde44de68e81c3e6923e5fd45b0295ad3/lib/token-store/cursors.js#L55-L56).

```js
const token = cursors.forward.createBaseCursor(
    this.tokens,
    this.comments,
    this.indexMap,
    offset, // startLoc
    -1,     // endLoc
    includeComments
).getOneToken()
```